elec/src/LMR16030_regulator.ato

```diff
@@ -19,54 +20,130 @@ component LMR16030SDDAR:
+
     # Power Input
     power_in.vcc ~ ic.VIN
     power_in.gnd ~ ic.GND
 
+    # 7.3.7 External Soft Start
+    c_ss = new Capacitor
+    ic.SS ~ c_ss.p1; c_ss.p2 ~ ic.GND
+    c_ss.value = soft_start_time * 3uA / 0.75V
+    # c_ss.footprint = "C0402"
+
     # 8.2.2.2 Output Voltage Set-Point
     # The remainder of this is handeled in our super class
     power_out ~ feedback_div.power
     feedback_div.output.vcc ~ ic.FB
+    v_ref = 0.75V
 
     # 8.2.2.3 Switching Frequency
     r_switching_frequency = new Resistor  # Rt in the datasheet
-    r_switching_frequency.value = 42904 * (switching_frequency / 1kHz)  ** -1.088
-    r_switching_frequency.package = "0402"
+    ic.RT_slash_SYNC ~ r_switching_frequency.p1; r_switching_frequency.p2 ~ ic.GND
+    # Use an assertion here rather than an assignment in case we replace the resistor elsewhere
+    assert r_switching_frequency.value within (42904 * (switching_frequency / 1kHz)  ** -1.088) * 1kΩ
+    r_switching_frequency.footprint = "C0402"
 
     # 8.2.2.4 Output Inductor Selection
     inductor = new Inductor
+    ic.SW ~ inductor.p1; inductor.p2 ~ power_out.vcc
     inductor_ripple_current = v_out * (v_in - v_out) / (v_in * inductor.value * switching_frequency)  # The ripple current in the inductor
+    k_ind = inductor_ripple_current / max_output_current
+    assert k_ind within 0.2 to 0.4
     # TODO: do we need to constrain the ripple current in some way?
-    assert inductor.value > (v_in - v_out) / (max_i_current * (0.2 to 0.4)) * v_out / (v_in * switching_frequency)
+    assert inductor.value > (v_in - v_out) / (max_output_current * k_ind) * v_out / (v_in * switching_frequency)
 
     # 8.2.2.5 Output Capacitor Selection
     c_out = new Capacitor
-    d_v_out_esr = (0.2 to 0.4) * max_i_current * c_out.esr
+    power_out ~ c_out.power
+    d_v_out_esr = inductor_ripple_current * c_out.esr
+    d_v_out_c = inductor_ripple_current / (8 * c_out.value * switching_frequency)
+
+    assert d_v_out_esr + d_v_out_c < output_ripple
+    # Since the above is generally a conservative case, we're
+    # ignoring the step change sizing case
+
+    # 8.2.2.6 Schottky Diode Selection
+    diode = new SchottkyDiode
+    ic.SW ~ diode.cathode; diode.anode ~ power_in.gnd
+    assert diode.voltage > v_in * 1.25  # 25% higher than max input voltage
+    assert diode.current > max_output_current
+
+    # 8.2.2.7 Input Capacitor Selection
+    c_in_bulk = new Capacitor
+    power_in ~ c_in_bulk.power
+    c_in_bulk.value = 4.7uF to 10uF
+    c_in_bulk.voltage = v_in * 2
+    c_in_bulk.dielectric = "X7R"
+    c_in_bulk.footprint = "C01005"
 
     c_in_hf = new Capacitor
     power_in ~ c_in_hf.power
+    c_in_hf.value = 100nF +/- 21%
     c_in_hf.voltage = v_in * 2
+    c_in_hf.footprint = "C01005"
+    c_in_hf.dielectric = "X7R"
 
-
+    # 8.2.2.8 Bootstrap Capacitor Selection
     c_boot = new Capacitor
+    ic.SW ~ c_boot.p1; c_boot.p2 ~ ic.BOOT
+    c_boot.value = 100nF +/- 21%
+    c_boot.voltage = 16V
```

Comment:
c_boot.voltage > 16V would make more sense. We should also standardize the variables for those generic components so we can propagate them to the server for selection and for them to show up with the language server.

Reply:
Yes, generic attributes and how they’re processed 100% needs standardization.

When I wrote this I originally thought that the component selector was being smart and taking this as the min voltage, but you’re right and I don’t actually think it’s going to.

We could consider introducing ‘inf’ (infinity) so we can represent ranged values as “above” and “below” some value. One fancier would be to take the syntax you suggested and interpret “>16uF” as “16uF to inf”